src/@types/InteractionContext.d.ts

```diff
@@ -112,6 +99,29 @@ declare global {
 		) => Promise<Message | boolean>;
 	}
 
+	interface BaseCommandContext extends InteractionContext {
+		/** The command invocation interaction. */
+		readonly interaction: CommandInteraction;
```

Comment:
With `skipLibCheck` disabled, TypeScript complains about this field not matching the one defined in the parent interface. Not sure the best way to fix this, since I don't know this new button abstraction well at all. Might need to decide whether our commands need a `CommandInteraction` or a `RepliableInteraction` in order to work properly, and use that for both. That or somehow decouple `BaseCommandContext` from `InteractionContext`, perhaps by creating a new parent context interface, or simplifying `BaseCommandContext` and raising that to be the parent interface instead.

It seems we only use `BaseCommandContext` internally in this file (except in the `talk` command, which could use a more specific interface like `RepliableInteraction` instead.

Reply:
This ended up being pretty simple to resolve by just using `BaseInteraction` as the type for interaction on `InteractionContext`. Typescript is smart enough to figure the rest out.